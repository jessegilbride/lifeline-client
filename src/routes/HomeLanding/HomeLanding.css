#homepage-banner {
  height: calc(100vh - var(--top-spacing) - 2rem);
  /* background-image: url('../../images/homepage-banner.jpg'); */
  background-repeat: no-repeat;
}

#homepage-banner h1 {
  margin-block-start: -6rem;
  padding: 1rem;
  font-size: 2.4rem;
  font-weight: normal;
  text-align: center;
  line-height: 1;
}

h1 span.bi-line {
  display: block;
  padding-top: 2rem;
  font-size: smaller;
  font-style: italic;
}

.feature {
  background-color: lightsalmon;
  color: black;
}

.site-intro-section {
  padding: 1rem 1rem;
  background-color: var(--eggplant);
  border-top: 3px solid var(--yellow);
  border-bottom: 3px solid var(--green);
  text-align: center;
}

.site-intro-content {
  font-size: 150%;
  line-height: 1.5;
  color: var(--mint);
}

@media screen and (min-width: 48em) {
  #homepage-banner {
    height: calc(100vh - var(--top-spacing) - 2rem);
  }
  #homepage-banner h1 {
    /* margin-block-start: -2rem; */
    font-size: 3.2rem;
  }
  .site-intro-section {
    padding-left: 4rem;
    padding-right: 4rem;
  }
}