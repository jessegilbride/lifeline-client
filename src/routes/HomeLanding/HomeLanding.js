import React, { Component, Fragment } from 'react';
// import TimelineContext from '../../contexts/TimelineContext';
import './HomeLanding.css';

export default class HomeLanding extends Component {
  // static contextType = TimelineContext;

  render() {
    return (
      <Fragment>
        <section className='page-top-banner' id='homepage-banner'>
          <h1 className=''>welcome</h1>
        </section>
        <section className='site-intro-section'>
          <p className='site-intro-content'>sub-section</p>
        </section>

      </Fragment>
    );
  }
}
