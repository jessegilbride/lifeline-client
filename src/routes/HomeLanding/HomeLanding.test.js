import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import HomeLanding from './HomeLanding';

describe('HomeLanding component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <MemoryRouter><HomeLanding /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <MemoryRouter><HomeLanding name="HomeLanding" /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});
