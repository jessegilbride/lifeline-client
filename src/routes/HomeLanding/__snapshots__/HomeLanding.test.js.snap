// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeLanding component renders the UI as expected 1`] = `
Array [
  <section
    className="page-top-banner"
    id="homepage-banner"
  >
    <h1
      className=""
    >
      welcome
    </h1>
  </section>,
  <section
    className="site-intro-section"
  >
    <p
      className="site-intro-content"
    >
      sub-section
    </p>
  </section>,
]
`;
